### PR TITLE
Return `CORPROF_E_PROFILER_CANCEL_ACTIVATION` if Tracing is Disabled

### PR DESCRIPTION
**Pull request by link04**
## Summary of changes

Changed return from `E_FAIL` to `CORPROF_E_PROFILER_CANCEL_ACTIVATION` for scenarios where the tracer is intentionally disabled by either the code or the user setting `DD_TRACE_ENABLED` set to `false`.

## Reason for change

To avoid printing misleading error event logs when there is not actual error.

## Test coverage

Tested locally using the command below to build the tracer locally:
`.\tracer\build.ps1 BuildTracerHome BuildNativeLoader PackageTracerHome`

After doing so I simply compared the outcome of disabling the tracer setting `DD_TRACE_ENABLED=false` in the tracer with the changes vs the one without them:

- Returning `E_FAIL`:
Image 2022-04-21 at 2 28 58 PM

- Returning `CORPROF_E_PROFILER_CANCEL_ACTIVATION`:
Image 2022-04-21 at 2 29 05 PM

NOTE: Also uninstalled & reinstalled the tracer and so no issues in the Event Logs or the in test service.

### CODE REVIEW

File: tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
     {
         Logger::Warn("DATADOG TRACER DIAGNOSTICS - Profiler disabled: .NET 5.0 runtime or greater is required on this "
                      "architecture.");
-        return E_FAIL;
+        return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:79]: I think this one could still be a "fail". Users tried to load the tracer, but we don't support the current environment.

File: tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 2]
```diff
@@ -152,7 +152,7 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         if (!functions_worker_runtime_value.empty() && !IsAzureFunctionsEnabled())
         {
             Logger::Info("DATADOG TRACER DIAGNOSTICS - Profiler explicitly disabled for Azure Functions.");
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:154]: @pierotibou: I thought auto-instrumentation worked in Azure Functions. Is this check wrong?

**Comment by lucaspimentel** [comment]
Can you add the "Test coverage" section back into the PR description and explain how you tested this?Even if it was completely manual tests. Thanks!

File: tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp (hunk 2 above)
Comment (zacharycmontoya) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:154]: The `IsAzureFunctionsEnabled()` function basically does looks for the `DD_TRACE_AZURE_FUNCTIONS_ENABLED` environment variable and returns `true` if not set, so this is fine

File: tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp (hunk 1 above)
Comment (zacharycmontoya) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:79]: +1, we'd want to raise an error in the event log for this scenario

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2685 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/ed74cf794cb02fb698567052caae973870b82428) will **not change** line coverage
:heavy_check_mark: Merging #2685 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/ed74cf794cb02fb698567052caae973870b82428) will **not change** branch coverage
:heavy_check_mark: Merging #2685 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/ed74cf794cb02fb698567052caae973870b82428) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/ed74cf794cb02fb698567052caae973870b82428) | #2685       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `13533` / `18627` | `13475` / `18627` |          |
| Lines %   | `73%`      | `72%`      |  `0%` :heavy_check_mark:  |
| Branches  | `7832` / `11125` | `7795` / `11125` |          |
| Branches %| `70%`      | `70%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `12431`      | `12431`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=96251&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/96251/artifacts?artifactName=Code%20Coverage%20Report_96251&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2685](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=96316&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/96316/artifacts?artifactName=Code%20Coverage%20Report_96316&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/ed74cf794cb02fb698567052caae973870b82428) | #2685       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `72%`       |  `0%` :heavy_check_mark:  |
| Branches %| `70%`      | `70%`       |  `0%` :heavy_check_mark:  |
| Complexity| `12431`      | `12431`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Telemetry.JsonWebRequestTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/b9333dfe775dfb9bc6840b1b8590c4cc67d60b62/tracer/src/Datadog.Trace/Telemetry/Transports/JsonWebRequestTelemetryTransport.cs) | `-27%` :no_entry: | `-57%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Telemetry.JsonHttpClientTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/b9333dfe775dfb9bc6840b1b8590c4cc67d60b62/tracer/src/Datadog.Trace/Telemetry/Transports/JsonHttpClientTelemetryTransport.cs) | `-26%` :no_entry: | `-38%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/b9333dfe775dfb9bc6840b1b8590c4cc67d60b62/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `-17%` :no_entry: | `-11%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.DiagnosticListeners.AspNetCoreDiagnosticObserver](https://github.com/DataDog/dd-trace-dotnet/tree/b9333dfe775dfb9bc6840b1b8590c4cc67d60b62/tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs) | `-4%` :warning: | `-6%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.CIVisibility](https://github.com/DataDog/dd-trace-dotnet/tree/b9333dfe775dfb9bc6840b1b8590c4cc67d60b62/tracer/src/Datadog.Trace/Ci/CIVisibility.cs) | `7%` :heavy_check_mark: | `7%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Processors.OriginTagTraceProcessor](https://github.com/DataDog/dd-trace-dotnet/tree/b9333dfe775dfb9bc6840b1b8590c4cc67d60b62/tracer/src/Datadog.Trace/Ci/Processors/OriginTagTraceProcessor.cs) | `9%` :heavy_check_mark: | `17%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=96251&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/96251/artifacts?artifactName=Code%20Coverage%20Report_96251&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2685](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=96316&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/96316/artifacts?artifactName=Code%20Coverage%20Report_96316&api-version=6.0&%24format=zip)

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2685 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc): 
* 3 benchmarks are **slower**, with geometric mean 1.277
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`WriteAndFlushEnrichedTraces`|net472|965μs|5.53μs|41μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|739μs|4μs|29.7μs|0| 0|0|2.57&nbsp;KB|
|#2685|`WriteAndFlushEnrichedTraces`|net472|972μs|5.31μs|30μs|0| 0|0|3.16&nbsp;KB|
|#2685|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|737μs|4.05μs|29.2μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`AllCycleSimpleBody`|net472|306ns|1.68ns|9.2ns|0.052| 0|0|329&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`AllCycleSimpleBody`|netcoreapp3.1|380ns|1.79ns|6.95ns|0.00452| 0|0|328&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`AllCycleMoreComplexBody`|net472|291ns|1.61ns|9.4ns|0.0482| 0|0|305&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`AllCycleMoreComplexBody`|netcoreapp3.1|379ns|2.02ns|10.1ns|0.00421| 0|0|304&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`BodyExtractorSimpleBody`|net472|433ns|2.48ns|19.5ns|0.057| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`BodyExtractorSimpleBody`|netcoreapp3.1|448ns|2.34ns|11ns|0.00368| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`BodyExtractorMoreComplexBody`|net472|26.5μs|132ns|590ns|1.18| 0.0121|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|21.1μs|115ns|651ns|0.093| 0|0|6.75&nbsp;KB|
|#2685|`AllCycleSimpleBody`|net472|291ns|1.33ns|4.97ns|0.052| 0|0|329&nbsp;B|
|#2685|`AllCycleSimpleBody`|netcoreapp3.1|382ns|1.53ns|5.92ns|0.00455| 0|0|328&nbsp;B|
|#2685|`AllCycleMoreComplexBody`|net472|283ns|1.1ns|4.12ns|0.0482| 0|0|305&nbsp;B|
|#2685|`AllCycleMoreComplexBody`|netcoreapp3.1|362ns|1.55ns|6.21ns|0.00414| 0|0|304&nbsp;B|
|#2685|`BodyExtractorSimpleBody`|net472|425ns|2.09ns|8.85ns|0.0569| 0|0|361&nbsp;B|
|#2685|`BodyExtractorSimpleBody`|netcoreapp3.1|438ns|2.09ns|8.38ns|0.00364| 0|0|272&nbsp;B|
|#2685|`BodyExtractorMoreComplexBody`|net472|24.3μs|122ns|545ns|1.19| 0.0119|0|7.62&nbsp;KB|
|#2685|`BodyExtractorMoreComplexBody`|netcoreapp3.1|21.1μs|115ns|688ns|0.0936| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`SendRequest`|netcoreapp3.1|302μs|1.48μs|6.63μs|0.151| 0|0|19.74&nbsp;KB|
|#2685|`SendRequest`|net472|0.00143ns|0.00104ns|0.00542ns|0| 0|0|0&nbsp;b|
|#2685|`SendRequest`|netcoreapp3.1|304μs|1.62μs|8.42μs|0.148| 0|0|19.74&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`ExecuteNonQuery`|net472|1.76μs|7.78ns|28.1ns|0.0942| 0|0|594&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`ExecuteNonQuery`|netcoreapp3.1|1.54μs|7.26ns|29ns|0.00906| 0|0|632&nbsp;B|
|#2685|`ExecuteNonQuery`|net472|1.79μs|7.35ns|27.5ns|0.0936| 0.000867|0|594&nbsp;B|
|#2685|`ExecuteNonQuery`|netcoreapp3.1|1.51μs|5.5ns|21.3ns|0.00889| 0|0|632&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`CallElasticsearch`|net472|2.65μs|13.9ns|66.7ns|0.126| 0|0|802&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`CallElasticsearch`|netcoreapp3.1|1.62μs|7.19ns|28.7ns|0.0105| 0|0|792&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`CallElasticsearchAsync`|net472|3.13μs|8.63ns|32.3ns|0.148| 0.00146|0|939&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`CallElasticsearchAsync`|netcoreapp3.1|1.81μs|9.37ns|42.9ns|0.0124| 0|0|912&nbsp;B|
|#2685|`CallElasticsearch`|net472|2.7μs|11.3ns|40.7ns|0.127| 0|0|803&nbsp;B|
|#2685|`CallElasticsearch`|netcoreapp3.1|1.64μs|7.8ns|32.2ns|0.0107| 0|0|792&nbsp;B|
|#2685|`CallElasticsearchAsync`|net472|2.92μs|9.83ns|35.5ns|0.149| 0|0|939&nbsp;B|
|#2685|`CallElasticsearchAsync`|netcoreapp3.1|1.96μs|9.78ns|41.5ns|0.0123| 0|0|912&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`ExecuteAsync`|net472|3.32μs|18.5ns|119ns|0.165| 0|0|1.06&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`ExecuteAsync`|netcoreapp3.1|2.03μs|8.57ns|30.9ns|0.0139| 0|0|1.03&nbsp;KB|
|#2685|`ExecuteAsync`|net472|3.17μs|13.8ns|53.6ns|0.167| 0.00155|0|1.06&nbsp;KB|
|#2685|`ExecuteAsync`|netcoreapp3.1|2.02μs|10.5ns|51.5ns|0.0149| 0|0|1.03&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`SendAsync`|net472|7.09μs|40.3ns|307ns|0.356| 0|0|2.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`SendAsync`|netcoreapp3.1|5.24μs|28.7ns|170ns|0.0307| 0|0|2.21&nbsp;KB|
|#2685|`SendAsync`|net472|6.98μs|39.6ns|286ns|0.354| 0|0|2.28&nbsp;KB|
|#2685|`SendAsync`|netcoreapp3.1|4.66μs|21.4ns|90.9ns|0.0308| 0|0|2.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|net472|3.35μs|17.9ns|92.8ns|0.228| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|netcoreapp3.1|3μs|13.1ns|49ns|0.021| 0|0|1.53&nbsp;KB|
|#2685|`EnrichedLog`|net472|3.49μs|17.3ns|77.2ns|0.228| 0|0|1.45&nbsp;KB|
|#2685|`EnrichedLog`|netcoreapp3.1|3.11μs|11.6ns|43.5ns|0.0215| 0|0|1.53&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|net472|286μs|699ns|2.52μs|0.427| 0.142|0|4.33&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|netcoreapp3.1|232μs|1.11μs|4.3μs|0| 0|0|4.21&nbsp;KB|
|#2685|`EnrichedLog`|net472|292μs|1.46μs|6.38μs|0.427| 0.142|0|4.33&nbsp;KB|
|#2685|`EnrichedLog`|netcoreapp3.1|232μs|1.27μs|7.52μs|0| 0|0|4.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|net472|7.87μs|41.6ns|208ns|0.503| 0|0|3.23&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|netcoreapp3.1|7.08μs|30.9ns|119ns|0.0493| 0|0|3.6&nbsp;KB|
|#2685|`EnrichedLog`|net472|7.43μs|30.1ns|128ns|0.507| 0|0|3.23&nbsp;KB|
|#2685|`EnrichedLog`|netcoreapp3.1|6.57μs|21.4ns|80.2ns|0.0488| 0|0|3.6&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`SendReceive`|net472|2.3μs|12.3ns|64.9ns|0.16| 0.00116|0|1.01&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`SendReceive`|netcoreapp3.1|2.01μs|10.3ns|68.6ns|0.0144| 0|0|1.01&nbsp;KB|
|#2685|`SendReceive`|net472|2.32μs|12ns|61.4ns|0.16| 0.00113|0|1.01&nbsp;KB|
|#2685|`SendReceive`|netcoreapp3.1|2μs|10.6ns|53.9ns|0.0136| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|net472|6.15μs|33.1ns|184ns|0.292| 0|0|1.87&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`EnrichedLog`|netcoreapp3.1|5.25μs|27.5ns|140ns|0.0204| 0|0|1.49&nbsp;KB|
|#2685|`EnrichedLog`|net472|5.93μs|21.6ns|89.1ns|0.292| 0|0|1.87&nbsp;KB|
|#2685|`EnrichedLog`|netcoreapp3.1|5.17μs|28ns|168ns|0.0198| 0|0|1.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2685

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishScope&#8209;netcoreapp3.1 | 1.347 | 1,012.35 | 1,363.44 | 
| Benchmarks.Trace.SpanBenchmark.StartFinishSpan&#8209;netcoreapp3.1 | 1.299 | 872.46 | 1,133.26 | several?

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`StartFinishSpan`|net472|882ns|4.67ns|24.7ns|0.0716| 0|0|457&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`StartFinishSpan`|netcoreapp3.1|880ns|4.47ns|19.5ns|0.00635| 0|0|456&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`StartFinishScope`|net472|1.14μs|6.39ns|42.4ns|0.0846| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`StartFinishScope`|netcoreapp3.1|1.01μs|3.85ns|14.4ns|0.0076| 0|0|576&nbsp;B|
|#2685|`StartFinishSpan`|net472|886ns|4.59ns|22.9ns|0.0717| 0|0|457&nbsp;B|
|#2685|`StartFinishSpan`|netcoreapp3.1|1.16μs|18.3ns|183ns|0.00635| 0|0|456&nbsp;B|
|#2685|`StartFinishScope`|net472|1.17μs|6.65ns|55.2ns|0.0842| 0|0|538&nbsp;B|
|#2685|`StartFinishScope`|netcoreapp3.1|1.38μs|15.5ns|155ns|0.00772| 0|0|576&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2685

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.TraceAnnotationsBenchmark.RunOnMethodBegin&#8209;net472 | 1.189 | 1,276.26 | 1,518.11 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`RunOnMethodBegin`|net472|1.27μs|6.3ns|27.4ns|0.0841| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/5d88f2c33acc18bac27435a1ae0c559c1b0c8edc)|`RunOnMethodBegin`|netcoreapp3.1|1.16μs|4.81ns|18.6ns|0.00806| 0|0|576&nbsp;B|
|#2685|`RunOnMethodBegin`|net472|1.59μs|20.2ns|195ns|0.0831| 0|0|538&nbsp;B|
|#2685|`RunOnMethodBegin`|netcoreapp3.1|1.23μs|6.56ns|32.8ns|0.00748| 0|0|576&nbsp;B|
</details>

**Review by robertpi — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]
LGTM, please either do a test locally to validate the results (and update the PR description with details) or add an automated test to validate 👍🏼

File: tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp (hunk 2 above)
Comment (lucaspimentel) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:154]: My mistake. I read `IsAzureFunctionsEnabled()` as "are we running in Azure Functions?", similar to the check for Kudu above 🤦🏽‍♂️

**Review by lucaspimentel — APPROVED** [review]

**Pull request merged**